tools/bridge/tests/conftest.py

```diff
@@ -218,3 +221,56 @@ def home_bridge_contract(
     )
 
     return contract
+
+
+@pytest.fixture
+def write_config(tmp_path):
+    """returns a function that writes a config file"""
+
+    def write(config):
+        p = tmp_path / "config.toml"
+        with open(p, "w") as f:
+            f.write(config)
+        return str(p)
+
+    return write
+
+
+@pytest.fixture
+def minimal_config():
+    """return a minimal configuration"""
+
+    return """
+[foreign_chain]
+rpc_url = "http://localhost:9200"
+token_contract_address = "0x731a10897d267e19B34503aD902d0A29173Ba4B1"
+bridge_contract_address = "0xb4c79daB8f259C7Aee6E5b2Aa729821864227e84"
+
+[home_chain]
+rpc_url = "http://localhost:9100"
+bridge_contract_address = "0x771434486a221c6146F27B72fd160Bdf0eb1288e"
+
+[validator_private_key]
+raw = "0xb8dcbb8a564483279579e04bffacbd76f79df157cfbebed84079673b32d9e72f"
+"""
+
+
+@pytest.fixture
+def webservice_config():
+    """webservice part of the configuration"""
+    return """
+[webservice]
+enabled = true
+host = "127.0.0.1"
+port = 9500
+"""
+
+
+@pytest.fixture
+def load_config_from_string():
```

Comment:
Something I've noticed a while ago, this is a just a good example: What's the reason to make this a fixture and not just a regular module-level function? I have the feeling that in general we're using fixtures quite often to just for the convenience of passing them via parameters (most of the time test data) and less often to run proper setup code which I think is the original intention of fixtures. Not sure if this is actually a problem, though, but curious about others' opinions.

Reply:
yes, you're right. I had been thinking about that when I wrote it. One problem is that I didn't have a canonical module to put this into.
Having said that my first implementation of that fixture also used the write_config fixture. So, in that case it had to be a fixture.

The same is also true for the minimal_config fixture, which is just a string. This one previously was just a global variable inside one test module, but became a fixture when I needed it in another test module.
I will try to move some of those into a dedicated test module and see how it looks like (not in this PR).

I would say the other use of fixtures is to parametrize tests.